lib/msf/core/handler/reverse_hop_http.rb

```diff
@@ -241,34 +241,27 @@ def initialize(info = {})
   #
   # Generates and sends a stage up to the hop point to be ready for the next client
   #
-  def send_new_stage
-    conn_id = generate_uri_checksum(URI_CHECKSUM_CONN) + "_" + Rex::Text.rand_text_alphanumeric(16)
+  def send_new_stage(uri)
+    # try to get the UUID out of the existing URI
+    info = process_uri_resource(uri)
+    uuid = info[:uuid] || Msf::Payload::UUID.new
+
+    # generate a new connect
+    sum = uri_checksum_lookup(:connect)
+    conn_id = generate_uri_uuid(sum, uuid)
     url = full_uri + conn_id + "/\x00"
 
     print_status("Preparing stage for next session #{conn_id}")
-    blob = stage_payload
-    #
-    # Patch options into the payload
-    #
-    Rex::Payloads::Meterpreter::Patch.patch_passive_service!(blob,
-      :ssl            => ssl?,
-      :url            => url,
-      :expiration     => datastore['SessionExpirationTimeout'],
-      :comm_timeout   => datastore['SessionCommunicationTimeout'],
-      :ua             => datastore['MeterpreterUserAgent'],
-      :proxy_host     => datastore['PayloadProxyHost'],
-      :proxy_port     => datastore['PayloadProxyPort'],
-      :proxy_type     => datastore['PayloadProxyType'],
-      :proxy_user     => datastore['PayloadProxyUser'],
-      :proxy_pass     => datastore['PayloadProxyPass'])
-
-    blob = encode_stage(blob)
+    blob = stage_payload({
+      :uuid => uuid,
```

Comment:
New code should use 1.9 hash syntax for symbol keys:
```ruby
  uuid: uuid,
```

This also makes it easier to transition to keyword arguments later.